README.md

```diff
@@ -146,6 +148,86 @@ title,content
 My Title,Hello World!
 ```
 
+### Creating a custom Task
+
+TODO: Add generation instructions
+
+If you have a special use case requiring iteration over an unsupported resource, you can implement the `enumerator_builder` method instead. For example, you may need to iterate over external resources fetch from some API.
+
+This method should return an object responding to `enumerator(context:)` with an Enumerator, yielding pairs of `[item, item_cursor]`. In order for your enumerator to support resuming iteration part way through, you may use the `context.cursor`.
+
+You may optionally provide an implementation for `count`, if appropriate.
```

Comment:
Maybe we should add a line about needing to implement `process` as well?

Reply:
I wasn't sure about that one, since it seems like a given. Maybe if I integrate it into the previous paragraph?

> ...yielding pairs of `[item, item_cursor]` **to the `process` method your `Task` defines.** In order for...

Reply:
Perfect!